### PR TITLE
Better READMEs + Vagrant scripts/documentation for quick testing

### PR DESCRIPTION
**Pull request by kmoppel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,24 @@
 PGObserver
 ==========
 
-PGObserver was developed at Zalando during the past years to monitor performance metrics of our different PostgreSQL clusters. Due to our use of a stored procedure API layer our strong focus was on monitoring that API layer. However the number of metrics was increased to include other relevant values:
+PGObserver is a monitoring solution for PostgreSQL databases, covering almost all metrics provided by the database engine's internal statistics collector.
+PGObserver works out of the box on all Postgres databases from version 9.0 (including AWS RDS) and does not require installing any server side database extensions nor privileged users, making it perfect for developers!
```
Comment (evakbe) [inline README.md]: ... version 9.0 (including AWS RDS) ..
sound like RDS has something to do with versions ..
better would be
on all Postgres databases (including AWS RDS) from version 9.0 ...

File: README.md [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+For some metrics though, data gathering wrapper functions (stored procedures) need to be installed on the server being monitored, to circumvent the superuser requirements.
```
Comment (evakbe) [inline README.md]: I recommend sticking with postgresql grammar i.e. ...data gathering functions (or stored procedures) have to be created on the server being monitored.

File: README.md [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
     - self time per procedure
+* All executed statements data (based on pg_stat_statements module, which needs to be enabled on the DB)
+    - query runtimes, call counts, time spent on IO
 * Table IO statistics
     - number of sequential and index scans
     - number of inserts, (hot) updates, deletes
     - table and index size
     - heap hits vs disk hits
-* CPU load
+* CPU load (needs a plpythonu stored procedure from sql/data_collection_helpers folder)
```
Comment (evakbe) [inline README.md]: this is the second "(requires/needs ...)"; maybe you should have a bullet point "*Requirement - blablabla..." to clearly structure these clauses

File: README.md [hunk 4]
```diff
@@ -23,18 +29,20 @@ Extra features ( require some additional setup, look into "extra_features" folde
...
+Gathering of single metrics is configurable per instance (see "hosts" table), so if you don't need so much details you don't have to gather them.
 
-The performance data is gathered by a Java application querying the different PostgreSQL performance views. Gather intervals for the different metrics can be configured on a per host per metric basis, enabling more detailed monitoring for critical systems and less details for less important systems, thus reducing the amount of data gathered. Additionally you can configure sets of hosts to monitor from different Java processes, e.g., if you deploy to multiple locations with limited accessability.
+Metrics are gathered by a Java application, querying the different PostgreSQL performance views (pg_stat_*).
+Gather intervals for the different metrics can be configured on a per host per metric basis, enabling more detailed monitoring for critical systems and less details for less important systems, thus reducing the amount of data gathered. Additionally you can configure sets of hosts to monitor from different Java processes, e.g., if you deploy to multiple locations with limited accessability.
```
Comment (evakbe) [inline README.md]: check spelling of "accessability"

File: README.md (hunk 2 above)
Comment (avaczi) [inline README.md]: To be precise, there are no stored procedures in PostgreSQL at all.  We use the term internally, yes, but it is not correct anyway :)  Let's stick with 'functions' until sprocs really appear.

File: README.md [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
+For some metrics though, data gathering wrapper functions (stored procedures) need to be installed on the server being monitored, to circumvent the superuser requirements.
 
-* Stored procedure data (pure sql procedures are not tracked)
+Monitored metrics include:
+
+* Stored procedure data
     - number of calls
     - run time per procedure
     - self time per procedure
+* All executed statements data (based on pg_stat_statements module, which needs to be enabled on the DB)
```
Comment (avaczi) [inline README.md]: This is contradicting to what you state above ('and does not require installing any server side database extensions')

File: README.md (hunk 3 above)
Comment (avaczi) [inline README.md]: and consequently unavailable for RDS.  The limitations of the single features should be clear.

File: README.md (hunk 4 above)
Comment (avaczi) [inline README.md]: Gather_ing_ intervals, maybe?

File: README.md [hunk 6]
```diff
@@ -23,18 +29,20 @@ Extra features ( require some additional setup, look into "extra_features" folde
...
+Metrics are gathered by a Java application, querying the different PostgreSQL performance views (pg_stat_*).
+Gather intervals for the different metrics can be configured on a per host per metric basis, enabling more detailed monitoring for critical systems and less details for less important systems, thus reducing the amount of data gathered. Additionally you can configure sets of hosts to monitor from different Java processes, e.g., if you deploy to multiple locations with limited accessability.
 
-The web frontend is a Python CherryPy standalone application. See the screenshot folder for basic examples. Chart visualization is rendered using JS flot library.
+The Web Frontend is a standalone Python + CherryPy application. See the "screenshots" folder for basic examples. Chart visualization is rendered using JS flot library.
```
Comment (avaczi) [inline README.md]: links would be welcome (both to the dir and http://www.flotcharts.org/  and http://www.cherrypy.org/

File: README.md [hunk 7]
```diff
@@ -23,18 +29,20 @@ Extra features ( require some additional setup, look into "extra_features" folde
...
 
 A testdata.py (frontends/src) script is included to generate minimalistic test data for a local test setup.
 
-Visit us at http://tech.zalando.com or feel free to create issues on Github
+Visit us at https://tech.zalando.com or feel free to create issues here on Github
```
Comment (avaczi) [inline README.md]: I am oldschool, I know, but https://tech.zalando.com/ (with the trailing slash), please :)

File: vagrant/start.sh [hunk 8]
```diff
@@ -18,13 +18,25 @@ done
 
 cd /vagrant/sql/schema
 psql -c "CREATE DATABASE $PGDATABASE;" postgres
-psql -c 'CREATE EXTENSION hstore;'
 find -name '*.sql' | sort | xargs cat | psql
-psql -f /vagrant/vagrant/initial.sql
 
-ip=$(ip -o -4 a show eth0|awk '{print $4}' | cut -d/ -f 1)
+psql -d "$PGDATABASE" -c "INSERT INTO hosts (host_name, host_user, host_password, host_db, host_ui_shortname, host_ui_longname) VALUES ('${PGHOST}', '${PGUSER}', '${PGPASSWORD}', '${PGDATABASE}', 'pgo', 'PgoMetricsDB')"
```
Comment (avaczi) [inline vagrant/start.sh]: these lines should be shorter

**Pull request merged**